File: repos_estudiantes_TARQUI_tercera_entrega/ILLANES_ESCOBAR_FERNANDO_JOHSUA/PERSISTENCIA/Ejercicio3/Python/gestor_cliente.py
import os
import pickle
from cliente import Cliente

class GestorCliente:
    def __init__(self, archivo="clientes.pkl"):
        self.archivo = archivo

    def guardar_cliente(self, cliente: Cliente):
        lista = self.leer_todos()
        lista.append(cliente)
        with open(self.archivo, 'wb') as f:
            pickle.dump(lista, f)

    def leer_todos(self):
        if not os.path.exists(self.archivo):
            return []
        with open(self.archivo, 'rb') as f:
            return pickle.load(f)

    def buscar_cliente(self, id: int):
        for c in self.leer_todos():
            if c.id == id:
                return c
        return None

    def buscar_celular_cliente(self, id: int):
        c = self.buscar_cliente(id)
        if c:
            return f"Datos del cliente: {c}\nCelular: {c.celular}"
        return "Cliente no encontrado."

File: repos_estudiantes_TARQUI_tercera_entrega/ILLANES_ESCOBAR_FERNANDO_JOHSUA/PERSISTENCIA/Ejercicio3/Java/GestorCliente.java
package PERSISTENCIA.Ejercicio3.Java;

import java.io.*;
import java.util.ArrayList;

public class GestorCliente {
    private final String archivo = "clientes.dat";

    public void guardarCliente(Cliente c) {
        ArrayList<Cliente> lista = leerTodos();
        lista.add(c);
        try (ObjectOutputStream out = new ObjectOutputStream(new FileOutputStream(archivo))) {
            out.writeObject(lista);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public ArrayList<Cliente> leerTodos() {
        try (ObjectInputStream in = new ObjectInputStream(new FileInputStream(archivo))) {
            return (ArrayList<Cliente>) in.readObject();
        } catch (IOException | ClassNotFoundException e) {
            return new ArrayList<>();
        }
    }

    public Cliente buscarCliente(int id) {
        for (Cliente c : leerTodos()) {
            if (c.getId() == id) return c;
        }
        return null;
    }

    public String buscarCelularCliente(int id) {
        Cliente c = buscarCliente(id);
        if (c != null) {
            return "Datos del cliente: " + c + "\nCelular: " + c.getCelular();
        }
        return "Cliente no encontrado.";
    }
}

